node/network/protocol/src/lib.rs

```diff
@@ -159,35 +159,49 @@ impl<M> NetworkBridgeEvent<M> {
 	}
 }
 
-/// A succinct representation of a peer's view. This consists of a bounded amount of chain heads.
+/// A succinct representation of a peer's view. This consists of a bounded amount of chain heads
+/// and the highest known finalized block number.
 ///
 /// Up to `N` (5?) chain heads.
 #[derive(Default, Debug, Clone, PartialEq, Eq, Encode, Decode)]
-pub struct View(pub Vec<Hash>);
+pub struct View {
+	/// A bounded amount of chain heads.
+	pub heads: Vec<Hash>,
+	/// The highest known finalized block number.
+	pub finalized_number: BlockNumber,
+}
 
 impl View {
+	/// Construct a new view with the given chain heads and finalized number 0.
+	pub fn new(heads: Vec<Hash>) -> Self {
+		Self {
+			heads,
+			finalized_number: 0,
```

Comment:
I don't think that this is safe to do so; are you sure that there is no potential harm for instance to the av-store pruning stuff?

Reply:
This function is only used for tests and is documented. Could add a note this is test only, or even replace with a macro.